bindings_ffi/src/mls.rs

```diff
@@ -28,6 +29,26 @@ use xmtp_mls::{
 
 pub type RustXmtpClient = MlsClient<TonicApiClient>;
 
+struct StoreManager {
+    store: Option<Arc<Mutex<EncryptedMessageStore>>>,
```

Comment:
We probably don't want so many Mutexes, too many of these will lead to deadlocks that will be difficult to debug. Diesel Connections/Pools already have internal locking mechanisms, so it shouldn't be necessary